src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -766,6 +763,19 @@ protected DynamoModel(IStartConfiguration config)
             }
         }
 
+        public bool AddPackagePath(string path)
+        {
+            if (!Directory.Exists(path))
+                return false;
+          
+            if (PreferenceSettings.CustomPackageFolders.Contains(path))
+                return false;
```

Comment:
these two if conditions can be  one  statement.

Reply:
They can be, but I separated them so that the logic was clearer. Why is making them one statement better?

Reply:
Nested if's are faster, because the conditions executes sequentially. if the results are different, then I would go with mutliple if's.

Reply:
This is not true. In a test application that duplicates the above logic both separated and combined and runs the comparison 100,000,000 times, both take 4304ms. The thought that combining the statements is faster may have been true in the past, but the code is optimized down to being the same in modern compilers.

Reply:
Thats not true. say there is a function that has to traverse 10000 data set to make a decision. 
writing like this has some cost
```
if(fooo()) return false;
if(foo) return false;
```
what if foo is already false, but the fooo() gets executed?

Reply:
That's order of testing, and is unaffected by whether the statements are combined or not.  You put the most frequently false and/or cheapest to test first.

Reply:
also, this should be about maintainability, this code gets called via a user command once every hour or something - the speed is totally irrelevant.